src/filesystem/impls/filer/BracketsFiler.js

```diff
@@ -3,16 +3,107 @@
 define(function (require, exports, module) {
     "use strict";
 
-    // If you need to debug Filer for some reason, drop the .min below
+    // TODO: we shouldn't be loading this just to get Path, Buffer, etc.
     var Filer = require("thirdparty/filer/dist/filer.min");
-    var fs;
+    var FilerBuffer = Filer.Buffer;
+    var ChannelUtils = require("thirdparty/MessageChannel/ChannelUtils");
+    var fnQueue = require("filesystem/impls/filer/lib/queue");
+    var UUID = ChannelUtils.UUID;
+    var port;
 
-    Filer.fs = function() {
-        if(!fs) {
-            fs = new Filer.FileSystem({provider: new Filer.FileSystem.providers.Memory()});
+    // Remote filesystem callbacks
+    var callbackQueue = {};
+
+    function remoteFSCallbackHandler(e) {
+        var data = e.data;
+        var callbackItem = callbackQueue[data.callback];
+        if(!callbackItem.persist) {
+            delete callbackQueue[data.callback];
+        }
+        callbackItem.callback.apply(null, data.result);
+    }
+
+    function receiveMessagePort(e) {
+        var data = e.data;
+        try {
+            data = JSON.parse(data);
+            data = data || {};
+        } catch(err) {
+            data = {};
+        }
+
+        if (data.type === "bramble:filer") {
+            window.removeEventListener("message", receiveMessagePort, false);
```

Comment:
third param default is `false` isn't it? so just `window.removeEventListener("message", receiveMessagePort)` should be fine?